keras/applications/convnext.py

```diff
@@ -522,7 +525,9 @@ def ConvNeXt(
         cur += depths[i]
 
     if include_top:
-        x = Head(num_classes=classes, name=model_name)(x)
+        x = Head(num_classes=classes,
+                 classifier_activation=classifier_activation,
+                 name=model_name)(x)
         imagenet_utils.validate_activation(classifier_activation, weights)
```

Comment:
Thanks for the fix, let's put the validation before building the head.

Reply:
Thanks, I thought I moved it yesterday. Also added a unit test for this case.

Reply:
As a second thought, we can also check the case where it is "linear", let me know if you want me to include this in unit test also.